internal/codegen/values.go

```diff
@@ -61,39 +66,88 @@ type config struct {
 	// Name is the name of this repository
 	Name string
 
+	// Deprecated: Versions should be treated as an argument via stencil.Arg instead
 	// Versions are versions of applications to be used over baked in defaults
 	Versions map[string]string
 }
 
-// IDEA(jaredallard): Allow extensions to provide values here? Or
-// do we just allow them to be called directly? Future consideration.
+// module contains information about the current module that
+// is rendering a template.
+type module struct {
+	// Name is the name of the current module
+	Name string
+
+	// Version is the version of the current module
+	Version string
+}
+
+// modulesSlice is a list of modules with helpers on top of it
+type modulesSlice []module
+
+// ByName returns a module by name
+func (m modulesSlice) ByName(name string) module {
+	for _, mod := range m {
+		if mod.Name == name {
+			return mod
+		}
+	}
+
+	return module{}
+}
 
 // Values is the top level container for variables being
 // passed to a stencil template.
 type Values struct {
 	// Git is information about the current git repository, if there is one
-	Git *git
+	Git git
```

Comment:
Made these not points to make `Copy` easy to implement, really was no reason to have these be pointers anyways